File: Code/Core/Graphics/Platform/DX/GPUBuffer.cpp
#include "GPUBuffer.h"

size_t Core::GPUBuffer::Size() const
{
    return size_t();
}

bool Core::GPUBuffer::IsWritable() const
{
    return false;
}

bool Core::GPUBuffer::IsReadable() const
{
    return false;
}

bool Core::GPUBuffer::WriteData(const void* data, size_t size)
{
    return false;
}

bool Core::GPUBuffer::WriteDataForTexture(const void* data, uint32_t width, uint32_t height, uint32_t formatSize)
{
    return false;
}

bool Core::GPUBuffer::ReadData(void* data, size_t size)
{
    return false;
}

File: Code/WMFramework/Graphics/Private/Metal/GPUBuffer.m
//
//  GPUBuffer.m
//  WMFramework
//
//  Created by Eugene Kim on 2023/04/29.
//

#import <Foundation/Foundation.h>
#include "Graphics/Private/Metal/GPUBuffer.h"

using namespace WildMini;

GPUBuffer::GPUBuffer(id<MTLBuffer> buffer)
{
    mtlBuffer = [buffer retain];
}

GPUBuffer::~GPUBuffer()
{
    [mtlBuffer release];
}

size_t GPUBuffer::Size() const
{
    return mtlBuffer.length;
}

bool GPUBuffer::IsWritable() const
{
    return false;
}

bool GPUBuffer::IsReadable() const
{
    return false;
}

bool GPUBuffer::WriteData(const void *data, size_t size)
{
    return false;
}

bool GPUBuffer::WriteDataForTexture(const void *data, uint32_t width, uint32_t height, uint32_t formatSize)
{
    return false;
}

bool GPUBuffer::ReadData(void *data, size_t size)
{ 
    return false;
}
